modules/integration-tests/src/test/java/org/wso2/carbon/apimgt/rest/integration/tests/store/APISubscriptionTestCaseIT.java

```diff
@@ -0,0 +1,179 @@
+        subscription = new Subscription().apiIdentifier(TestUtil.getApi("baseapi1").getId())
+                .applicationId(application.getApplicationId()).policy("Gold");
+        subscription = subscriptionIndividualApi.subscriptionsPost(subscription);
+    }
+
+    @Test(dependsOnMethods = "testCreateSubscription")
+    public void testGetAllSubscriptionsByApplication() throws AMIntegrationTestException {
+        SubscriptionCollectionApi subscriptionCollectionApi = TestUtil.getStoreApiClient("user1", TestUtil.getUser
+                ("user1"), AMIntegrationTestConstants.DEFAULT_SCOPES).buildClient(SubscriptionCollectionApi.class);
+        SubscriptionList subscriptionList = subscriptionCollectionApi.subscriptionsGet("", application
+                .getApplicationId(), "", 0, 10, "");
+        Assert.assertNotNull(subscriptionList);
+        Assert.assertEquals(subscriptionList.getCount().intValue(), 1);
+        Subscription subscription = subscriptionList.getList().get(0);
+        Assert.assertEquals(subscription.getApiIdentifier(), this.subscription.getApiIdentifier());
+        Assert.assertEquals(subscription.getPolicy(), this.subscription.getPolicy());
+    }
+
+    @Test(dependsOnMethods = "testCreateSubscription")
+    public void testGetAllSubscriptionsByAPIId() throws AMIntegrationTestException {
+        SubscriptionCollectionApi subscriptionCollectionApi = TestUtil.getStoreApiClient("user1", TestUtil.getUser
+                ("user1"), AMIntegrationTestConstants.DEFAULT_SCOPES).buildClient(SubscriptionCollectionApi.class);
+        SubscriptionList subscriptionList = subscriptionCollectionApi.subscriptionsGet(subscription.getApiIdentifier
+                (), "", "", 0, 10, "");
+        Assert.assertNotNull(subscriptionList);
+        Assert.assertEquals(subscriptionList.getCount().intValue(), 1);
+        Subscription subscription = subscriptionList.getList().get(0);
+        Assert.assertEquals(subscription.getApplicationId(), this.subscription.getApplicationId());
+        Assert.assertEquals(subscription.getPolicy(), this.subscription.getPolicy());
+    }
+
+    @Test(dependsOnMethods = "testCreateApplication")
+    public void testCreateSubscriptionWIthNonExistingPolicy() {
+        String apiId = TestUtil.getApi("baseapi1").getId();
+        String applicationId = application.getApplicationId();
+        Subscription subscriptionNegative = new Subscription().apiIdentifier(apiId).applicationId(applicationId)
+                .policy("Bronze1");
+        try {
+            subscriptionIndividualApi.subscriptionsPost(subscriptionNegative);
+            Assert.fail();
+        } catch (Exception e) {
+            if (e.getCause() instanceof RestAPIException) {
+                RestAPIException restAPIException = (RestAPIException) e.getCause();
+                Error error = restAPIException.getError();
+                Assert.assertEquals(error.getMessage(), "Policy Not found");
```

Comment:
This level of detail is good but if we do any changes to the formatting of the exception message it will lead to test failures. Its bets if we can use a constant to define the error message string and reuse the same in the code and test